osf/models/registrations.py

```diff
@@ -825,17 +824,9 @@ def withdraw(self):
         # Pass auth=None because the registration initiator may not be
         # an admin on components (component admins had the opportunity
         # to disapprove the retraction by this point)
-        for node in self.get_descendants_recursive(primary_only=True):
+        for node in self.node_and_primary_descendants():
             node.set_privacy('public', auth=None, log=False)
-            node.update_search()
-
-        # force a save before sending data to share or retraction will not be updated
```

Comment:
I'm guessing the behavior described here was changed on the SHARE side, and therefore this block is now irrelevant?

Reply:
the comment is not wrong, but it's not a special case here (should always save before calling `update_search`) -- what happened here is that all the nodes should be set public before updating search (because they reference each other as `hasPart`/`isPartOf`), hence `bulk_update_search`; also i realized `set_privacy` does a save (so the explicit/called-out save is unnecessary); and then @jwalz suggested using `node_and_primary_descendants` to dedupe this logic, which i thought made sense